source/fuzz/fact_manager.cpp

```diff
@@ -696,9 +737,78 @@ void FactManager::DataSynonymAndIdEquationFacts::AddDataSynonymFactRecursive(
   assert(DataDescriptorsAreWellFormedAndComparable(context, dd1, dd2));
 
   // Record that the data descriptors provided in the fact are equivalent.
+  // Both |dd1| and |dd2| are registered in the equivalence relation after this
+  // point.
   MakeEquivalent(dd1, dd2);
 
-  // We now check whether this is a synonym about composite objects.  If it is,
+  // Compute various corollary facts.
+  ComputeConversionSynonymFacts(dd1, dd2, context);
+  ComputeCompositeDataSynonymFacts(dd1, dd2, context);
+}
+
+void FactManager::DataSynonymAndIdEquationFacts::ComputeConversionSynonymFacts(
+    const protobufs::DataDescriptor& dd1, const protobufs::DataDescriptor& dd2,
```

Comment:
dd2 doesn't get used except in an assertion.

Thus please change this so that it takes a single data descriptor, `dd`.  It can then compute dd's representative and look for multiple conversions on members of that class.

Reply:
Fixed in d7d0f49.